### ISSUE TITLE
Support for types implementing multiple interfaces

### ISSUE THREAD

**Issue by darrellwarde** [neo4j/graphql#1102]
As it stands, schema generation can fail when a type implements more than one interface which is used as the type of a relationship field.

For example, this example from issue #1085 causes an error:

```gql
type Address {
  id: ID! @id
  locates: [WithAddress!]! @relationship(type: "ADDRESS", direction: OUT) 
}

type OrganisationNumber {
  value: String! @unique
  identifies: [WithOrganisationIdentifier!]! @relationship(type: "ORGNR", direction: OUT) 
}

interface WithAddress {
  address: Address! @relationship(type: "ADDRESS", direction: IN) 
}

interface WithOrganisationIdentifier {
  orgNr: OrganisationNumber! @relationship(type: "ORGNR", direction: IN) 
}

type Controller implements WithAddress & WithOrganisationIdentifier {
  name: String!
  address: Address! @relationship(type: "ADDRESS", direction: IN) 
  orgNr: OrganisationNumber! @relationship(type: "ORGNR", direction: IN) 
}

type Processor implements WithAddress & WithOrganisationIdentifier {
  name: String!
  address: Address! @relationship(type: "ADDRESS", direction: IN) 
  orgNr: OrganisationNumber! @relationship(type: "ORGNR", direction: IN) 
}
```

**Comment by Hashdhi** [comment]
Any tentative timeline to Support for types implementing multiple interfaces i.e Abstract Types 2.0 milestone release date ?

**Comment by darrellwarde** [comment]
Hey @Hashdhi, we have it on our roadmap to looks at the architecture of abstract types as a whole in about the middle of next year, not too close I'm afraid. We have a major release that we need to develop and release first before we can move onto this.

**Comment by Hashdhi** [comment]
Thanks Darell

**Comment by darrellwarde** [comment]
This should be resolved in the upcoming 5.0.0 release.
